MyEtherWallet/MyEtherWallet
# Use network unit everywhere, fix network redux state

Closes #726.

This originally was only meant to have the confirmation show ETC or whatever instead of ETH when on a different network, but turns out I kind of pulled up the floorboards on `state.config.network` in the process.

The problem was that `selectors/config`'s `getNetworkConfig` was having to fetch which network we were on each time, rather than just using `state.config.network`. If you made the mistake of accessing the latter, it would just be whatever network the site bootstrapped with. Worse yet, the site was always bootstrapping with ETH, even if the starting node was an ETC node or otherwise.

The full list of changes are as follows:

* Confirmation modal now shows correct network token (Unless ERC20 token, then it shows that)
* Send unit dropdown now shows correct network token (Instead of "ether" for all networks)
* Changing node now correctly updates `state.config.network`
* Loading saved node from local storage now correctly sets `state.config.network` at bootstrap

All in all, I think this PR is a good example of doing _more_ in sagas, and _less_ in selectors. Selectors should ideally not be creating new objects or doing overly complex logic, but instead just be selecting things off of the redux store.

## Screenshots

<img width="484" alt="screen shot 2018-01-09 at 12 50 35 pm" src="https://user-images.githubusercontent.com/649992/34734987-d00f0128-f53b-11e7-867f-3947de9d124f.png">

<img width="647" alt="screen shot 2018-01-09 at 12 51 17 pm" src="https://user-images.githubusercontent.com/649992/34734992-d25c94f4-f53b-11e7-987b-e071250d4f78.png">


## Steps to Test

1. Load up the site connected to an ETH node
2. Go to send and unlock
3. Confirm the dropdown says ETH
4. Switch to an ETC node
5. Confirm the dropdown now says ETC
6. Refresh the page with ETC as your network
7. Confirm that your page starts connected to the ETC node
8. Confirm that send shows ETC in the dropdown
9. Generate a transaction, and check that the modal says ETC
10. Switch back to an ETH node
11. Generate a transaction, and check that the modal says ETH
12. Generate a token transaction, and check that the modal says the token name
